### PR TITLE
ARC kconfig change to use CONFIG_INIT_ARCH_HW_AT_BOOT

### PR DESCRIPTION
**Pull request by YuguoWH**
According to #29400 which introduces a new kconfig option CONFIG_INIT_ARCH_HW_AT_BOOT, align kconfig option CONFIG_ARC_CUSTOM_INIT to it.

### CODE REVIEW

**Review by ioannisg — APPROVED** [review]
Looks fine to me. I guess this should be taken out of draft as soon as #29400 is merged.

**Comment by YuguoWH** [comment]
rebase and open for review

File: arch/arc/core/reset.S [hunk 1]
```diff
@@ -83,7 +83,7 @@ SECTION_FUNC(TEXT,__start)
 	sr r0, [_ARC_V2_IRQ_VECT_BASE]
 #endif
 
-#if defined(CONFIG_USERSPACE)
+#ifdef CONFIG_USERSPACE
```
Comment (evgeniy-paltsev) [inline arch/arc/core/reset.S:86]: Hi Yuguo,
I'm wondering how is this change related to this commit/PR topic?

If it is cleanup, could you please move it to separate patch, thanks.

File: arch/arc/core/reset.S [hunk 2]
```diff
@@ -176,7 +177,7 @@ _slave_core_wait:
 	/* signal master core that slave core runs */
 	st 0, [arc_cpu_wake_flag]
 
-#if defined(CONFIG_ARC_FIRQ_STACK)
+#ifdef CONFIG_ARC_FIRQ_STACK
```
Comment (evgeniy-paltsev) [inline arch/arc/core/reset.S:180]: Same here

File: arch/arc/core/reset.S [hunk 3]
```diff
@@ -205,7 +206,7 @@ _master_core_startup:
 	mov_s sp, INIT_STACK
 	add sp, sp, INIT_STACK_SIZE
 
-#if defined(CONFIG_ARC_FIRQ_STACK)
+#ifdef CONFIG_ARC_FIRQ_STACK
```
Comment (evgeniy-paltsev) [inline arch/arc/core/reset.S:209]: Same here

**Review by abrodkin — CHANGES_REQUESTED** [review]
Please do it all (except clean-up related to `#ifdef CONFIG_USERSPACE`) in 1 commit as this is really an atomic change.

**Review by abrodkin — APPROVED** [review]
I'd refer to an earlier commits which:
1. Initially introduced `ARC_CUSTOM_INIT` and
2. `CONFIG_INIT_ARCH_HW_AT_BOOT` later which takes over `ARC_CUSTOM_INIT` functionality and thus paving a way for code unification between multiple architectures.

That will help us in the future to understand logic behind those changes.

**Review by ruuddw — APPROVED** [review]

**Review by evgeniy-paltsev — APPROVED** [review]
